Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+  //  bocList: The BocList to which the row belongs.
+  //  currentRow: The row that fired the click event.
+  //  selectorControl: The selection selectorControl in this row.
+  //  returns: true if the row selection has changed.
+  public static OnRowClick(evt: JQueryEventObject, bocList: HTMLElement, currentRow: HTMLElement, selectorControl: JQuery): boolean
+  {
+    if (BocList._isCommandClick)
+    {
+      BocList._isCommandClick = false;
+      return false;
+    }  
+  
+    var currentRowBlock = new BocList_RowBlock (currentRow, selectorControl);
+    var selectedRows = BocList._selectedRows[bocList.id]
+    var isRowHighlightingEnabled = true
+    var isCtrlKeyPress = false;
+    if (evt)
+      isCtrlKeyPress = evt.ctrlKey;
+      
+    if (   selectedRows?.Selection == BocList._rowSelectionUndefined
+        || selectedRows?.Selection == BocList._rowSelectionDisabled)
+    {
+      return false;
+    }
+      
+    if (isCtrlKeyPress || BocList._isRowSelectorClickExecuting)
+    {
+      //  Is current row selected?
+      if (selectedRows?.Rows[selectorControl.id] != null)
+      {
+        //  Remove currentRow from list and unselect it
+        BocList.UnselectRow (bocList, currentRowBlock, isRowHighlightingEnabled);
+      }
+      else
+      {
+        if (  (   selectedRows?.Selection == BocList._rowSelectionSingleCheckBox
+               || selectedRows?.Selection == BocList._rowSelectionSingleRadioButton)
+            && selectedRows?.Length > 0)
+        {
+          //  Unselect all rows and clear the list
+          BocList.UnselectAllRows (bocList, isRowHighlightingEnabled);
+        }
+        //  Add currentRow to list and select it
+        BocList.SelectRow (bocList, currentRowBlock, isRowHighlightingEnabled);
+      }
+    }
+    else // cancel previous selection and select a new row
+    {
+      if (selectedRows != undefined && selectedRows.Length > 0) //TODO what to do here if selectedRows is undefined? should we throw an error before we even get here?
+      {
+        //  Unselect all rows and clear the list
+        BocList.UnselectAllRows (bocList, isRowHighlightingEnabled);
+      }
+      //  Add currentRow to list and select it
+      BocList.SelectRow (bocList, currentRowBlock, isRowHighlightingEnabled);
+    }
+    try
+    {
+      selectorControl.focus();
+    }
+    catch (e)
+    {
+    }  
+    BocList._isRowSelectorClickExecuting = false;
+    return true;
+  }
+
+  //  Selects a row.
+  //  Adds the row to the _bocList_selectedRows array and increments _bocList_selectedRowsLength.
+  //  bocList: The BocList to which the row belongs.
+  //  rowBlock: The row to be selected.
+  //  isRowHighlightingEnabled: true to mark the selected rows via css-class
+  public static SelectRow(bocList: HTMLElement, rowBlock: BocList_RowBlock, isRowHighlightingEnabled: boolean): void
+  {
+    //  Add currentRow to list  
+    var selectedRows = BocList._selectedRows[bocList.id];
```

Comment:
Here we can also just assume that the DOM structure is correct and use a `!` to override nullability here in the declaration of `selectedRows`. We can then remove the `if` checks again.